openpype/modules/deadline/abstract_submit_deadline.py

```diff
@@ -381,23 +364,35 @@ def serialize(self):
 
         """
         def filter_data(a, v):
-            if a.name.startswith("_"):
+            if isinstance(v, (DeadlineIndexedVar, DeadlineKeyValueVar)):
                 return False
             if v is None:
                 return False
             return True
 
         serialized = attr.asdict(
             self, dict_factory=OrderedDict, filter=filter_data)
-        serialized.update(self.EnvironmentKeyValue)
-        serialized.update(self.ExtraInfo)
-        serialized.update(self.ExtraInfoKeyValue)
-        serialized.update(self.TaskExtraInfoName)
-        serialized.update(self.OutputFilename)
-        serialized.update(self.OutputFilenameTile)
-        serialized.update(self.OutputDirectory)
+
+        # Custom serialize these attributes
+        for attribute in [
+            self.EnvironmentKeyValue,
+            self.ExtraInfo,
+            self.ExtraInfoKeyValue,
+            self.TaskExtraInfoName,
+            self.OutputFilename,
+            self.OutputFilenameTile,
+            self.OutputDirectory,
+            self.AssetDependency
+        ]:
+            serialized.update(attribute.serialize())
```

Comment:
Does anyone know if it's possible to implement a method on a class so `attr.as_dict` knows how to serialize it? E.g. a `__json__` method or alike?

Reply:
I guess the right way would be to pass [`value_serializer` to `attr.todict](https://www.attrs.org/en/stable/extending.html?highlight=serialize#customize-value-serialization-in-asdict)